modules/core/src/main/scala/fs2/kafka/internal/KafkaConsumerActor.scala

```diff
@@ -276,7 +286,7 @@ private[kafka] final class KafkaConsumerActor[F[_], K, V](
     def pollConsumer(state: State[F, K, V]): F[ConsumerRecords] =
       withConsumer
         .blocking { consumer =>
-          val assigned = consumer.assignment.toSet
+          val assigned = state.assignments
```

Comment:
Could you please explain how your changes address the issue?  The current codebase seems it doesn't have any flaw with managing assigned partitions.